ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -1538,9 +1537,9 @@ private static HdfsDirSnapshot addToSnapshot(Map<Path, HdfsDirSnapshot> dirToSna
   public static Map<Path, HdfsDirSnapshot> getHdfsDirSnapshots(final FileSystem fs, final Path path)
```

Comment:
I am not sure if this function is used by the Cleaner in the current upstream code. AFAIK Cleaner uses `getHdfsDirSnapshotsForCleaner` for getting the map of snapshots containing `HdfsDirSnapshot`.
Link for the function `getHdfsDirSnapshotsForCleaner` - https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java#L1501

On the cleaner side, the function is used here - 
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/handler/CompactionCleaner.java#L269

So from the Cleaner perspective, changing `getHdfsDirSnapshots` might not be very useful.

Reply:
@SourabhBadhya Thanks for the input. But I could see the other API also getting used here
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/handler/CompactionCleaner.java#L270

The below is the call hierarchy:
![image](https://user-images.githubusercontent.com/59791497/224981787-42a163dd-a925-43f0-be71-edf653f37f6b.png)

Reply:
Yeah the call hierarchy is true, but this is triggered only when the map is `null` in the `getAcidState()` call. In the case of Cleaner, the map of `HdfsDirSnapshots` is already created in prior and is used to call `getAcidState()`.
Relevant code with the null condition -https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java#L1350-L1352

Other usecases of getting the acid state of a table might hit this codepath. But it seems like Cleaner will never hit this codepath.

Reply:
@SourabhBadhya and @veghlaci05 Now I changed the logic same as _getHdfsDirSnapshotsForCleaner_ API(now all tests are passing). Even the Cleaner is not using this API same issue can happen for others so I corrected it. Please review and merge the code changes.